WSRequestHandler.cpp

```diff
@@ -1343,6 +1345,141 @@ void WSRequestHandler::HandleSetSceneItemCrop(WSRequestHandler* req) {
 }
 
 /**
+ * Gets the scene specific properties of the specified source item.
+ *
+ * @param {String (optional)} `scene-name` the name of the scene that the source item belongs to. Defaults to the current scene.
+ * @param {String} `item` The name of the source.
+ *
+ * @return TO POPULATE
+ *
+ * @api requests
+ * @name GetSceneItemSceneProperties
+ * @category sources
+ * @since 4.1.3?
+ */
+void WSRequestHandler::HandleGetSceneItemSceneProperties(WSRequestHandler* req) {
+    if (!req->hasField("item")) {
+        req->SendErrorResponse("missing request parameters");
+        return;
+    }
+
+    const char* item_name = obs_data_get_string(req->data, "item");
+    if (!str_valid(item_name)) {
+        req->SendErrorResponse("invalid request parameters");
+        return;
+    }
+
+    const char* scene_name = obs_data_get_string(req->data, "scene-name");
+    obs_source_t* scene = Utils::GetSceneFromNameOrCurrent(scene_name);
+    if (!scene) {
+        req->SendErrorResponse("requested scene doesn't exist");
+        return;
+    }
+
+    obs_sceneitem_t* scene_item = Utils::GetSceneItemFromName(scene, item_name);
+    if (scene_item) {
+        obs_data_t* data = obs_data_create();
+
+        // is name even needed here?
+        obs_data_set_string(data, "name", item_name);
+        
+        // is nested ideal?
+        obs_data_t* pos_data = obs_data_create();
+        vec2 pos;
+        obs_sceneitem_get_pos(scene_item, &pos);
+        obs_data_set_double(pos_data, "x", pos.x);
+        obs_data_set_double(pos_data, "y", pos.y);
+        obs_data_set_obj(data, "position", pos_data);
+
+        obs_data_set_double(data, "rotation", obs_sceneitem_get_rot(scene_item));
+
+        // investigate this and maybe convert so string in switch statment
+        // #define OBS_ALIGN_CENTER (0)
+        // #define OBS_ALIGN_LEFT   (1<<0)
+        // #define OBS_ALIGN_RIGHT  (1<<1)
+        // #define OBS_ALIGN_TOP    (1<<2)
+        // #define OBS_ALIGN_BOTTOM (1<<3)
+        obs_data_set_int(data, "alignment", obs_sceneitem_get_alignment(scene_item));
```

Comment:
Alignment is an `int` that corresponds to the above `enum` definition. Should I break this out into a string based on cases or just offload this responsibility to next programmer?